feat(pyroscope.ebpf): Add no_kernel_version_check argument (#6475)
grafana/alloy: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md

Reviewer: ```suggestion

{{< admonition type="caution" >}}
Use `no_kernel_version_check` only when you run on an older kernel that includes backported eBPF features from a later release.
When you skip this check, the profiler can't verify that your kernel has the eBPF features it needs, and profiling may fail or produce incomplete results.
Proceed at your own risk.
{{< /admonition >}}
```

Maybe something like this?

| `go_table_fallback`       | `bool`                   | Deprecated (no-op), previously enabled symbol lookup in `.sym` / `.dynsym` sections when `.gopclntab` lookup failed. | `false`  | no       |
| `hotspot_enabled`         | `bool`                   | A flag to enable or disable HotSpot profiling.                                                                       | `true`   | no       |
| `kernel_frames`           | `bool`                   | Include kernel-space frames in collected profiles. Set to `false` to drop kernel frames from each stack trace.       | `true`   | no       |

> | `no_kernel_version_check` | `bool`                   | Disable checking the kernel version for eBPF support. Use at your own risk, to run on older kernels with backported eBPF features. | `false`  | no       |

Reviewer: ```suggestion
| `no_kernel_version_check` | `bool`                   | Skip the kernel version check for eBPF support.                                                                      | `false`  | no       |
```

We can move the risk statement out of the table and into a caution admonition.
